File: src/HashCode.java
package org.nlogo.extensions.string;

import org.nlogo.api.Argument;
import org.nlogo.api.Context;
import org.nlogo.api.DefaultReporter;
import org.nlogo.api.ExtensionException;
import org.nlogo.api.LogoException;
import org.nlogo.api.Syntax;

public class HashCode extends DefaultReporter
{
    // take one string as input, reports the integer hash code
    
    public Syntax getSyntax()
    {
        return Syntax.reporterSyntax(
            new int[] {Syntax.StringType()}, Syntax.NumberType()
        ) ;
    }
    
    public Object report(Argument args[], Context context)
        throws ExtensionException, LogoException
    {
             
        // use typesafe helper method from 
        // org.nlogo.api.Argument to access argument
        // get string from args 0, get hashcode from that, convert
        String s = args[0].getString() ;
        Double n = new Double( s.hashCode() ) ;
        String s2 = n.toString() ;
        
        
        return n  ;

    }
}

File: src/HashCode.scala
package org.nlogo.extensions.string

import org.nlogo.api.Argument
import org.nlogo.api.Context
import org.nlogo.api.Reporter
import org.nlogo.api.ExtensionException
import org.nlogo.api.LogoException
import org.nlogo.core.Syntax

class HashCode extends Reporter
{
    // take one string as input, reports the integer hash code

    def getSyntax(): Syntax =
    {
      Syntax.reporterSyntax(
        right = List(Syntax.StringType),
        ret   = Syntax.NumberType)
    }

    @throws(classOf[ExtensionException])
    @throws(classOf[LogoException])
    def report(args: Array[Argument], context: Context): Object =
    {
        val ds: Double = args(0).getString.hashCode
        ds.asInstanceOf[AnyRef]
    }
}
